### PR TITLE
RSDK-3599 tp space path quality

### PR DESCRIPTION
**Pull request by biotinker**
While working on this PR it became clear that a real solution for TP-space path quality involves removing the precompute and instead enabling actual FK/IK on the PTGs. This PR does not do that.

This PR simplifies some of the tp-space code, and makes an important change for improving the quality of generated paths; specifically, in the case where a PTG is extended to its limit towards a goal, it will do another round of querying to try to reach that same goal.

To clarify the above: if a goal is an unobstructed 10 meters in front of a base, but PTGs are only precomputed to 4 meters, then this would have only extended towards the goal 4 meters, then gone to RRT random tree extension, thus virtually guaranteeing that the straight-line path directly to the goal was not found, favoring something more convoluted. This tweak allows the tree to extend to 4m, then to 8m, then to the goal, with zero random iterations in this example.

A smoothing algorithm was also prototyped as part of this work but was removed prior to the submission of this PR as it was not found to be suitable for use under the current system.

This PR will hopefully also solve RSDK-4050

Pinging @tessavitabile @kkufieta @nthakur7

### CODE REVIEW

File: motionplan/tpSpaceRRT.go [hunk 1]
```diff
@@ -302,62 +305,103 @@ func (mp *tpSpaceRRTMotionPlanner) getExtensionCandidate(
...
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+		}
+		candidates := []*candidate{}
+		for ptgNum, curPtg := range mp.tpFrame.PTGs() {
+			// Find the best traj point for each traj family, and store for later comparison
+			// TODO: run in parallel
+			cand := mp.getExtensionCandidate(ctx, goalNode, ptgNum, curPtg, rrt, seedNode, mp.planOpts)
```
Comment (raybjork) [inline motionplan/tpSpaceRRT.go:326]: Why not return an error separate from the candidate? Also, if there is an error we seem to just continue on is that what we want?

**Review by raybjork — CHANGES_REQUESTED** [review]

File: referenceframe/ptgFrame.go [hunk 2]
```diff
@@ -92,7 +92,7 @@ func (pf *ptgGridSimFrame) Transform(inputs []Input) (spatialmath.Pose, error) {
 	lastPose := spatialmath.NewZeroPose()
 	for _, trajNode := range traj {
 		// Walk the trajectory until we pass the specified distance
-		if trajNode.Dist > inputs[distanceAlongTrajectoryIndex].Value {
+		if trajNode.Dist < inputs[distanceAlongTrajectoryIndex].Value {
```
[the remark this answers is not in the document]
Comment (raybjork) [reply referenceframe/ptgFrame.go:95]: I think doing this is a lot more readable and you can simplify to a singular return statement if you would prefer only one by switching it to a `break`

File: motionplan/tpspace/ptgGridSim.go [hunk 3]
```diff
@@ -183,23 +176,12 @@ func (ptg *ptgGridSim) simulateTrajectories(simPtg PrecomputePTG) ([][]*TrajNode
 			dist += v * ptg.diffT
 			t += ptg.diffT
 
-			wpDist1 := math.Sqrt(math.Pow(wpX-x, 2) + math.Pow(wpY-y, 2))
-			wpDist2 := math.Abs(wpPhi - phi)
-			wpDist := math.Max(wpDist1, wpDist2)
-
-			if wpDist > ptg.minDist {
-				// If our waypoint is farther along than our minimum, update
+			// Update velocities of last node because reasons
```
[the remark this answers is not in the document]
Comment (raybjork) [reply motionplan/tpspace/ptgGridSim.go:179]: Mostly was just wondering what the reasons were, but I'm not overly concerned by it

**Review by raybjork — APPROVED** [review]
Would like to see the nit changed but up to you.  LGTM either way

File: referenceframe/ptgFrame.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (biotinker) [reply referenceframe/ptgFrame.go:95]: True, will do that.

**Review by nfranczak — APPROVED** [review]
🚀

**Pull request merged**